Tools/clinic/clinic.py

```diff
@@ -959,28 +968,17 @@ def parser_body(prototype, *fields, declarations=''):
                 if p.is_vararg():
                     if not new_or_init:
                         parser_code.append(normalize_snippet("""
-                            %s = PyTuple_New(%s);
-                            if (!%s) {{
-                                goto exit;
-                            }}
-                            for (Py_ssize_t i = 0; i < %s; ++i) {{
-                                PyTuple_SET_ITEM(%s, i, Py_NewRef(args[%d + i]));
-                            }}
+                            %s = %s + %d;
                             """ % (
-                                p.converter.parser_name,
-                                left_args,
-                                p.converter.parser_name,
-                                left_args,
-                                p.converter.parser_name,
-                                max_pos
-                            ), indent=4))
+                            p.converter.parser_name,
+                            p.name,
```

Comment:
I would expect here `p.converter.name` instead.

I'm trying your patch for converting some vararg functions in the math module (see the [old conversion attempt](https://github.com/python/cpython/pull/101124/commits/2ffe808a7b618019db8bf51ca8ee8d8298e4e5b6)), and it seems that without this change autogenerated code is broken, because it references a Python parameter name instead of C (i.e., `coordinates`, instead of `args`):
```c
static PyObject *
math_hypot_impl(PyObject *module, Py_ssize_t varargssize,
                PyObject *const *args);

static PyObject *
math_hypot(PyObject *module, PyObject *const *args, Py_ssize_t nargs)
{
    PyObject *return_value = NULL;
    Py_ssize_t varargssize = nargs;
    PyObject *const *__clinic_args;

    if (!_PyArg_CheckPositional("hypot", nargs, 0, PY_SSIZE_T_MAX)) {
        goto exit;
    }
    __clinic_args = coordinates + 0;
    return_value = math_hypot_impl(module, varargssize, __clinic_args);

exit:
    return return_value;
}
```
See [docs](https://docs.python.org/3/howto/clinic.html#renaming-the-c-functions-and-variables-generated-by-argument-clinic).  BTW, I think there should be a test for that.

Reply:
Hardcoding "args" will work too, but I'm not sure it's a good idea.

Reply:
Thanks for the advice.
When the function signature is `METH_FASTCALL`, the pass-in parameter is always named `args`, so replacing `p.name` with `args` works.
I made a fix and added two tests. It looks correct now.